taqueria-utils/taqueria-utils.ts

```diff
@@ -19,19 +19,64 @@ export const decodeJson = <T>(encoded: string): Future<TaqError, T> =>
 
 export const log = <T>(message: string) => (input: T) : T => {
     console.log(message)
-    console.log(input)
     return input
 }
 
+export const logInput = <T>(message: string) => (input: T) : T => {
+    console.log(message)
+    console.log(input)
+    return input
+}
 
 export const debug = <T>(input: T) => {
     debugger
     return input
 }
 
 const mkdirFuture = (path: string): Future<TaqError, void> => attemptP(() => Deno.mkdir(path, {recursive: true}))
+export const mkdir = (path: string) : Future<TaqError, string> => pipe(path, mkdirFuture, map(() => path))
 
-export const mkdir = (path: string) : Future<TaqError, string> => pipe(path, mkdirFuture, map (() => path))
+export const ensurePathExists = (path: string) : Future<TaqError, SanitizedAbsPath> => attemptP(async () =>{
+    try {
+        await Deno.stat(path);
+        return SanitizedAbsPath.create(path);
+    } catch(_e) {
+        return Promise.reject({ kind: 'E_INVALID_PATH_DOES_NOT_EXIST', msg: 'TODO i18n message' })
+    }  
+});
+
+export const ensurePathDoesNotExist = (path: string) : Future<TaqError, SanitizedAbsPath> => attemptP(async () =>{
+    try {
+        await Deno.stat(path);
+        return Promise.reject({ kind: 'E_INVALID_PATH_ALREADY_EXISTS', msg: 'TODO i18n message' })
+    } catch(_e) {
+        // Expect exception when trying to stat a new directory
+        return SanitizedAbsPath.create(path);
+    }
+});
+
+const rmFuture = (path: SanitizedAbsPath): Future<TaqError, void> => attemptP(async () => {
+    try {
+        await Deno.remove(path.value);
+    } catch {
+        // Ignore if path does not exist
+    }
+});
+export const rm = (path: SanitizedAbsPath) : Future<TaqError, SanitizedAbsPath> => pipe(path, rmFuture, map(() => path))
+
+export const gitClone = (url: SanitizedUrl) => (destinationPath: SanitizedAbsPath) => pipe(
+    attemptP(async () => {
+        const cloneProcess = Deno.run({
+            cmd: ["git", "clone", url.value, destinationPath.value],
+        });
+        const cloneResult = await cloneProcess.status();
+        
+        if (!cloneResult.success) {
+            return Promise.reject({kind: 'E_SCAFFOLD_URL_GIT_CLONE_FAILED', msg: 'TODO i18n message'})
```

Comment:
Related to a previous comment:
https://github.com/ecadlabs/taqueria/pull/139/files#r810071895